hail/src/main/scala/is/hail/expr/ir/lowering/LowerDistributedSort.scala

```diff
@@ -583,3 +587,76 @@ case class Chunk(filename: String, size: Long, byteSize: Long)
 case class SegmentResult(indices: IndexedSeq[Int], interval: Interval, chunks: IndexedSeq[Chunk])
 case class OutputPartition(indices: IndexedSeq[Int], interval: Interval, files: IndexedSeq[String])
 case class LoopState(largeSegments: IndexedSeq[SegmentResult], smallSegments: IndexedSeq[SegmentResult], readyOutputParts: IndexedSeq[OutputPartition])
+
+case class DistributionSortReader(key: TStruct, keyed: Boolean, spec: TypedCodecSpec, orderedOutputPartitions: IndexedSeq[OutputPartition], globals: IR, rowType: TStruct, rt: RTable) extends TableReader {
+  lazy val fullType: TableType = TableType(
+    rowType,
+    if (keyed) key.fieldNames else FastIndexedSeq(),
+    globals.typ.asInstanceOf[TStruct]
+  )
+
+  override def pathsUsed: Seq[String] = Seq()
```

Comment:
Short answer -- maybe

longer answer -- I don't think this really matters, pathsUsed is used to try to catch when users read/write to the same table file. Anything generated in the compiler that's not user-exposed doesn't actually need to define stuff here. I think the most correct answer for pathsUsed would be the root directory of all the output partitions (rather than each file individually). Should probably rename pathsUsed to `userPathsUsed`